### PR TITLE
Ports Opacity Refactor

### PR DESCRIPTION
**Pull request by Tsar-Salat**
Ports:
- https://github.com/tgstation/tgstation/pull/63877 (where some of the hardel stuff is stolen from)
- https://github.com/tgstation/tgstation/pull/52881

## About The Pull Request
Hi, this is a pretty light opacity refactor. In addition to the opacity refactor are some hardel fixes in atoms_movable

> - Moves all opacity var manipulation to a proc which sends a signal.
> - light_blocker element for movable opaque atoms made, which tracks its movement and updates the affected turfs for proper lighting updates.
> - has_opaque_atom boolean replaced by the opacity_sources lazylist to keep track of the sources, and a directional_opacity which serves a similar function but also allows for future expansion with on-border opaque objects (not yet implemented).
> - Some opacity-related sight procs optimized as a result of this.
> - Some variables moved to the object's definition.
> - A define or two added into the mix for clarity.
> - Some code cleaning, like turning booleans into their defines.

## Why It's Good For The Game
 There have been several port attempts here with this system, but they've all tried to include it as an umbrella as part of an "overlay lighting" or a "directional lighting" or whatever.

I don't really care about those. I'm not trying to port systems like those. Leave that to #8093 or something.

I need this refactor for Spatial Grid, and I need Spatial Grid to finish Smooth Movement.

Im getting the harddels out of the way now, so I dont have to deal with them in the next pr. 
Mech change done because its nonsensical if we are trying to turn mechs into proper vehicles.

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

Opacity functioning as expected.
opacity

</details>

## Changelog
:cl: RKz, Rohesie
refactor: minor opacity refactor
fix: Fixes some atoms_movable related harddels
balance: mechs no longer block sight
/:cl:

### CODE REVIEW

File: code/modules/lighting/lighting_turf.dm [hunk 1]
```diff
@@ -1,18 +1,3 @@
...
-	var/tmp/lighting_corners_initialised = FALSE
-
-	var/tmp/atom/movable/lighting_object/lighting_object // Our lighting object.
-	///Lighting Corner datums.
-	var/tmp/datum/lighting_corner/lighting_corner_NE
-	var/tmp/datum/lighting_corner/lighting_corner_SE
-	var/tmp/datum/lighting_corner/lighting_corner_SW
-	var/tmp/datum/lighting_corner/lighting_corner_NW
-
-	var/tmp/has_opaque_atom = FALSE // Not to be confused with opacity, this will be TRUE if there's any opaque atom on the tile.
```
Comment (itsmeow) [inline code/modules/lighting/lighting_turf.dm:14]: leave this stuff here please, it's better modularized like this rather than in the base turf.dm

**Review by itsmeow — CHANGES_REQUESTED** [review]

File: code/modules/lighting/lighting_turf.dm (hunk 1 above)
Comment (Tsar-Salat) [reply code/modules/lighting/lighting_turf.dm:14]: altered in 3b43bbf

**Review by itsmeow — APPROVED** [review]

File: tgui/package.json [hunk 2]
```diff
@@ -55,7 +55,7 @@
     "ts-plugin-inferno": "^1.1.0",
     "typescript": "^4.9.5",
     "url-loader": "^4.1.1",
-    "webpack": "^5.76.0",
+    "webpack": "^5.76.1",
```
Comment (itsmeow) [inline tgui/package.json:58]: yo what the fuck

Comment (PowerfulBacon) [reply tgui/package.json:58]: whoops

Comment (Tsar-Salat) [reply tgui/package.json:58]: Fuck

**Pull request merged**